tmt/base.py

```diff
@@ -1346,3 +1371,83 @@ def export(self):
         if self.note:
             data['note'] = self.note
         return data
+
+
+class Link(object):
+    """ Core attribute link parsing """
+
+    _relations = [
+        'verifies', 'verified-by',
+        'implements', 'implemented-by',
+        'documents', 'documented-by',
+        'blocks', 'blocked-by',
+        'duplicates', 'duplicated-by',
+        'parent', 'child',
+        'relates',
+        ]
+
+    def __init__(self, data=None):
+        """ Convert link data into canonical form """
+        # Nothing to do if no data provided
+        self.links = []
+        if data is None:
+            return
+        if not isinstance(data, list):
+            data = [data]
+
+        # Ensure that each link is in the canonical form
+        for link in data:
+
+            # It should be a string
+            if isinstance(link, str):
+                self.links.append(dict(relates=link))
+                continue
+
+            # Or a dictionary
+            if not isinstance(link, dict):
+                raise tmt.utils.SpecificationError(
+                    f"Invalid link target (should be 'str' or 'dict', "
+                    f"got '{type(link).__name__}'.")
+
+            # Verify the relation
+            relations = []
+            for key in link:
+                if key in ['url', 'ref', 'path', 'name', 'note']:
```

Comment:
Hm, 'note' should not be allowed as fmf identifier, right? IMO this should have a comment saying 'checking for fmf-id keys'